qiskit/tools/visualization/_counts_visualization.py

```diff
@@ -17,35 +17,68 @@
 import matplotlib.pyplot as plt
 
 
-def plot_histogram(data, number_to_keep=False):
+def plot_histogram(data, number_to_keep=False, legend=None, options=None):
```

Comment:
Can we emit deprecation warnings for `number_to_keep`? Change the signature of the function to:

```python
def plot_histogram(data, legend=None, options=None, **kwargs):
```

And check if `number_to_keep` has been passed. If so, emit a deprecation warning:

```python
if 'numbers_to_keep' in kwargs:
  warnings.warn('number_to_keep has been deprecated, use ...' DeprecationWarning)

numbers_to_keep = kwargs.get('numbers_to_keep', False)
```

Reply:
I don't want to change the signature here, because the order of named kwargs actually matters. For example someone could be using this like: `plot_histogram(data_input, 5)` and that would work fine but if I move numbers_to_keep those users will be broken. But, I'll add the deprecation warning